runtime/tests/checker/for_test.go

```diff
@@ -273,3 +274,122 @@ func TestCheckInvalidForShadowing(t *testing.T) {
 
 	assert.IsType(t, &sema.RedeclarationError{}, errs[0])
 }
+
+func TestCheckReferencesInForLoop(t *testing.T) {
+
```

Comment:
I see earlier that the references to the elements are created with unauthorized access, but can we add a couple tests for that behavior too when the outer array reference is entitled?

Reply:
Added: c659d79c67e50a58a2b1f981488c2be00858d77f